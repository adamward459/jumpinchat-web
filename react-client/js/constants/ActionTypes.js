/**
 * Created by Zaccary on 20/06/2015.
 */

export const SET_LAYOUT = 'SET_LAYOUT';
export const SET_SETTINGS_MODAL = 'SET_SETTINGS_MODAL';
export const SET_SETTINGS_ERROR = 'SET_SETTINGS_ERROR';


export const ROOM_LOAD = 'ROOM_LOAD';
export const ROOM_EMOJI = 'ROOM_EMOJI';
export const SET_ROOM_SETTINGS = 'SET_ROOM_SETTINGS';
export const SET_CLIENT_USER = 'SET_CLIENT_USER';
export const MESSAGE_ADD = 'MESSAGE_ADD';
export const MESSAGE_SEND = 'MESSAGE_SEND';
export const USER_CONNECT = 'USER_CONNECT';
export const USER_UPDATE_USERLIST = 'USER_UPDATE_USERLIST';
export const USER_DISCONNECT = 'USER_DISCONNECT';
export const USER_UPDATE_HANDLE = 'USER_UPDATE_HANDLE';
export const USER_UPDATE = 'USER_UPDATE';
export const USER_SET_NOTIFICATIONS = 'USER_SET_NOTIFICATIONS';
export const USER_SET_THEME = 'USER_SET_THEME';
export const USER_SET_UNREAD_MESSAGES = 'USER_SET_UNREAD_MESSAGES';
export const USER_SET_BROADCAST_QUALITY = 'USER_SET_BROADCAST_QUALITY';


export const SET_USER_LIST_OPTION_LIST = 'SET_USER_LIST_OPTION_LIST';
export const STREAM_REMOTE_DESTROY = 'STREAM_REMOTE_DESTROY';
export const STREAM_REMOTE_HANGUP = 'STREAM_REMOTE_HANGUP';
export const STREAM_REMOTE_RESUME = 'STREAM_REMOTE_RESUME';
export const STREAM_HANGUP_ALL = 'STREAM_HANGUP_ALL';
export const STREAM_RESUME_ALL = 'STREAM_RESUME_ALL';
export const STREAM_AUDIO_TOGGLE_MUTE = 'STREAM_AUDIO_TOGGLE_MUTE';
export const STREAM_AUDIO_TOGGLE_MUTE_ALL = 'STREAM_AUDIO_TOGGLE_MUTE_ALL';
export const STREAM_AUDIO_SET_GLOBAL_VOLUME = 'STREAM_AUDIO_SET_GLOBAL_VOLUME';
export const STREAM_LOCAL_CAN_BROADCAST = 'STREAM_LOCAL_CAN_BROADCAST';
export const STREAM_LOCAL_ADD = 'STREAM_LOCAL_ADD';
export const STREAM_LOCAL_DESTROY = 'STREAM_LOCAL_DESTROY';
export const STREAM_REMOTE_ADD = 'STREAM_REMOTE_ADD';
export const SET_STREAM_OPTIONS_STATE = 'SET_STREAM_OPTIONS_STATE';
export const SET_MESSAGE_SOUND = 'SET_MESSAGE_SOUND';
export const SET_SHOW_USERLIST = 'SET_SHOW_USERLIST';
export const SET_CHAT_INPUT_VALUE = 'SET_CHAT_INPUT_VALUE';
export const SET_WINDOW_IS_VISIBLE = 'SET_WINDOW_IS_VISIBLE';
export const SET_UNREAD_MESSAGES = 'SET_UNREAD_MESSAGES';
export const SET_CLIENT_PTT = 'SET_CLIENT_PTT';
export const SET_DEFAULT_PTT = 'SET_DEFAULT_PTT';
export const SET_REMOTE_STREAM_VOLUME = 'SET_REMOTE_STREAM_VOLUME';
export const STREAM_AUDIO_SET_ACTIVE = 'STREAM_AUDIO_SET_ACTIVE';
export const STREAM_SHOW_VOLUME_SLIDER = 'STREAM_SHOW_VOLUME_SLIDER';
export const STREAM_SET_LOADING = 'STREAM_SET_LOADING';
export const SET_LOCAL_AUDIO_ACTIVE = 'SET_LOCAL_AUDIO_ACTIVE';
export const RESUME_AUDIO_CONTEXT = 'RESUME_AUDIO_CONTEXT';
export const STREAM_SET_QUALITY = 'STREAM_SET_QUALITY';
export const SET_QUALITY_OPTIONS = 'SET_QUALITY_OPTIONS';
export const SET_MESSAGE_HISTORY = 'SET_MESSAGE_HISTORY';

// private messages
export const PM_START_CONVERSATION = 'PM_START_CONVERSATION';
export const PM_SELECT_CONVERSATION = 'PM_SELECT_CONVERSATION';
export const PM_REMOVE_CONVERSATION = 'PM_REMOVE_CONVERSATION';
export const PM_OPEN_MENU = 'PM_OPEN_MENU';
export const PM_CLOSE_CONVERSATION = 'PM_CLOSE_CONVERSATION';
export const PM_SET_CONVO_READ = 'PM_SET_CONVO_READ';

// Chat
export const CHAT_SET_USERS = 'CHAT_SET_USERS';
export const CHAT_SET_SETTINGS = 'CHAT_SET_SETTINGS';
export const CHAT_CLEAR_FEED = 'CHAT_CLEAR_FEED';
export const CHAT_SET_SCROLL = 'CHAT_SET_SCROLL';
export const CHAT_SET_SCROLL_FIXED = 'CHAT_SET_SCROLL_FIXED';

export const RESTORE_MESSAGE = 'RESTORE_MESSAGE';
export const SELECT_CHAT_TAB = 'SELECT_CHAT_TAB';
export const ADD_PRIVATE_MESSAGE = 'ADD_PRIVATE_MESSAGE';
export const SET_EMOJI_SEARCH = 'SET_EMOJI_SEARCH';
export const SET_EMOJI_RESULT_SELECTED = 'SET_EMOJI_RESULT_SELECTED';
export const SET_BANLIST = 'SET_BANLIST';

// modals
export const HANDLE_MODAL_STATE = 'HANDLE_MODAL_STATE';
export const BANLIST_MODAL_STATE = 'BANLIST_MODAL_STATE';
export const MEDIA_MODAL_STATE = 'MEDIA_MODAL_STATE';
export const MEDIA_MODAL_TYPE = 'MEDIA_MODAL_TYPE';
export const MEDIA_MODAL_SET_DEVICE_ID = 'MEDIA_MODAL_SET_DEVICE_ID';
export const MEDIA_MODAL_SET_AUDIO_PTT = 'MEDIA_MODAL_SET_AUDIO_PTT';
export const MEDIA_MODAL_LOADING = 'MEDIA_MODAL_LOADING';
export const MODAL_ERROR = 'MODAL_ERROR';
export const SET_JOIN_CONDITION_MODAL = 'SET_JOIN_CONDITION_MODAL';
export const SET_PROFILE_MODAL = 'SET_PROFILE_MODAL';
export const SET_IGNORE_MODAL = 'SET_IGNORE_MODAL';
export const SET_BAN_MODAL = 'SET_BAN_MODAL';


// sessions
export const SESSION_ID_SAVE = 'SESSION_ID_SAVE';
export const SESSION_IS_RECONNECTING = 'SESSION_IS_RECONNECTING';

export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
export const CLOSE_NOTIFICATION = 'CLOSE_NOTIFICATION';
export const PAUSE_NOTIFICATIONS = 'PAUSE_NOTIFICATIONS';
export const RESUME_NOTIFICATION = 'RESUME_NOTIFICATION';

export const CLIENT_UPDATE_HANDLE = 'CLIENT_UPDATE_HANDLE';

export const USER_SET = 'USER_SET';
export const USER_ACT_TOKEN_SET = 'USER_ACT_TOKEN_SET';
export const USER_LOG_IN = 'USER_LOG_IN';
export const HANDLE_MODAL_OPEN = 'HANDLE_MODAL_OPEN';
export const HANDLE_MODAL_CLOSE = 'HANDLE_MODAL_CLOSE';
export const HANDLE_MODAL_ERROR = 'HANDLE_MODAL_ERROR';
export const ACCOUNT_MODAL_STATE = 'ACCOUNT_MODAL_STATE';
export const INFO_MODAL_STATE = 'INFO_MODAL_STATE';
export const REPORT_MODAL_STATE = 'REPORT_MODAL_STATE';
export const SET_BROADCAST_RESTRICTED = 'SET_BROADCAST_RESTRICTED';
export const SET_EMOJI_PICKER = 'SET_EMOJI_PICKER';
export const INSERT_EMOJI = 'INSERT_EMOJI';
export const UPDATE_IGNORE = 'UPDATE_IGNORE';
export const USER_SET_PLAY_VIDEOS = 'USER_SET_PLAY_VIDEOS';

// Youtube
export const YOUTUBE_MODAL_STATE = 'YOUTUBE_MODAL_STATE';
export const YOUTUBE_SET_RESULTS = 'YOUTUBE_SET_RESULTS';
export const YOUTUBE_SET_VIDEO = 'YOUTUBE_SET_VIDEO';
export const YOUTUBE_SET_OPTIONS = 'YOUTUBE_SET_OPTIONS';
export const YOUTUBE_CLIENT_CLOSE_VIDEO = 'YOUTUBE_CLIENT_CLOSE_VIDEO';
export const YOUTUBE_ROOM_CLOSE_VIDEO = 'YOUTUBE_ROOM_CLOSE_VIDEO';
export const YOUTUBE_SET_VOLUME = 'YOUTUBE_SET_VOLUME';
export const YOUTUBE_SET_RESULTS_LOADING = 'YOUTUBE_SET_RESULTS_LOADING';
export const YOUTUBE_SET_VOLUME_CONTROL = 'YOUTUBE_SET_VOLUME_CONTROL';
export const YOUTUBE_SET_VIDEO_PAUSED = 'YOUTUBE_SET_VIDEO_PAUSED';
export const YOUTUBE_SET_VIDEO_RESUMED = 'YOUTUBE_SET_VIDEO_RESUMED';
export const YOUTUBE_SET_PLAYLIST = 'YOUTUBE_SET_PLAYLIST';
export const YOUTUBE_SET_PLAYLIST_ITEM_OPTIONS = 'YOUTUBE_SET_PLAYLIST_ITEM_OPTIONS';
export const YOUTUBE_CLEAR_SEARCH = 'YOUTUBE_CLEAR_SEARCH';
export const YOUTUBE_SET_VIDEO_SEEK = 'YOUTUBE_SET_VIDEO_SEEK';

// Push
export const PUSH_SET_MANAGER = 'PUSH_SET_MANAGER';
export const PUSH_SET_PUBLIC_KEY = 'PUSH_SET_PUBLIC_KEY';

export const PROFILE_SET = 'PROFILE_SET';
export const PROFILE_SET_NEW = 'PROFILE_SET_NEW';
export const PROFILE_LOADING = 'PROFILE_LOADING';
export const PROFILE_SET_IGNORE_ITEM = 'PROFILE_SET_IGNORE_ITEM';

// roles
export const ROLE_UPDATE = 'ROLE_UPDATE';
export const ROLES_SET = 'ROLES_SET';
export const ROLES_FETCH = 'ROLES_FETCH';
export const ROLES_SAVE = 'ROLES_SAVE';
export const ROLES_REMOVE = 'ROLES_REMOVE';
export const ENROLLMENTS_FETCH = 'ENROLLMENTS_FETCH';
export const ENROLLMENTS_SET = 'ENROLLMENTS_SET';
export const ENROLLMENT_USER_ADD = 'ENROLLMENT_USER_ADD';
export const ENROLLMENT_USER_ADD_FAILED = 'ENROLLMENT_USER_ADD_FAILED';
export const ENROLL_USER = 'ENROLL_USER';
export const UNENROLL_USER = 'UNENROLL_USER';
