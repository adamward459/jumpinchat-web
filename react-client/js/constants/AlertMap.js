export const ALERT_COLORS = {
  INFO: 'blue',
  SUCCESS: 'green',
  WARNING: 'yellow',
  ERROR: 'red',
};


export const ALERT_TYPES = {
  BANNER: 'banner',
  DIALOG: 'dialog',
  CHAT: 'chat',
};
