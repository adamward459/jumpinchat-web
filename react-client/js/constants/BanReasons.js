export default {
  BAN_BROADCAST_VIOLATION: 'Inappropriate broadcast',
  BAN_NUDITY: 'Nudity',
  BAN_RECORDING: 'Recording broadcasters',
  BAN_ABUSE: 'Abusing or promoting abuse',
  BAN_HARASSMENT: 'Harassing other members',
  BAN_UNDERAGE: 'Under age',
};
