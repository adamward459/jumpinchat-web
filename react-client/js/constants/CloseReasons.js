export const codes = {
  BAN_BROADCAST_VIOLATION: 'BAN_BROADCAST_VIOLATION',
  BAN_NUDITY: 'BAN_NUDITY',
  BAN_ABUSE: 'BAN_ABUSE',
  BAN_UNDERAGE: 'BAN_UNDERAGE',
};

export const reasons = {
  BAN_BROADCAST_VIOLATION: 'Inappropriate broadcasts',
  BAN_NUDITY: 'Nudity or encouraging nudity',
  BAN_ABUSE: 'Abusing or promoting abuse',
  BAN_UNDERAGE: 'Under age',
};
