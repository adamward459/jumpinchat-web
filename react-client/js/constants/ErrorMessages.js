/**
 * Created by Zaccary on 31/01/2016.
 */

export const ERR_NO_USER = 'no user found';
export const ERR_NO_ROOM = 'room does not exists';
export const ERR_NO_ROOM_ID = 'missing room ID';
export const ERR_NO_USER_LIST_ID = 'user not found in list';
export const ERR_NO_OP_TYPE = 'no moderator type';
export const ERR_NO_HANDLE = 'handle missing';
export const ERR_HANDLE_EXISTS = 'handle already in use';
export const ERR_HANDLE_VALIDATION = 'handle uses invalid characters';
export const ERR_HANDLE_LENGTH = 'handle too long';
export const ERR_USER_BANNED = 'banned';
export const ERR_NO_USER_SESSION = 'no user session';

export const ERR_YT_DEFAULT = 'Can\'t play video for some reason';
export const ERR_YT_BAD_PARAMS = 'Invalid parameters, can\'t play video';
export const ERR_YT_NO_HTML5 = 'Video can not be played in HTML5 player';
export const ERR_YT_NO_VIDEO = 'Video not found';
export const ERR_YT_NO_EMBED = 'Video can\'t be played while embedded';
