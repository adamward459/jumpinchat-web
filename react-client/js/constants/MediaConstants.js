export const defaultVideoConstraints = {
  width: 320,
  height: 240,
  frameRate: {
    ideal: 15,
    max: 30,
  },
};

export const goldVideoConstraints = {
  width: 960,
  height: 720,
  frameRate: {
    ideal: 30,
    max: 30,
  },
};

export const getVideoConstraints = config => ({
  width: {
    min: 320,
    ideal: config.dimensions.width,
    max: config.dimensions.width,
  },
  height: {
    min: 240,
    ideal: config.dimensions.height,
    max: config.dimensions.height,
  },
  frameRate: {
    ideal: config.frameRate,
    max: config.frameRate,
  },
});

export const goldVideoBitrate = 1024 * 1000;
