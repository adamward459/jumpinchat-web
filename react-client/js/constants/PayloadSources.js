export const SERVER_ACTION = 'SERVER_ACTION';
export const VIEW_ACTION = 'VIEW_ACTION';
