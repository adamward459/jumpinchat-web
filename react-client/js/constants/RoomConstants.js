/**
 * Created by Zaccary on 22/06/2015.
 */


export const RoomConstants = {
  maxMessages: 100,
};

export const chatTabs = {
  CHAT_FEED: 'CHAT_FEED',
  CHAT_PM: 'CHAT_PM',
  CHAT_USER_LIST: 'CHAT_USER_LIST',
};

export const layouts = {
  VERTICAL: 'VERTICAL',
  HORIZONTAL: 'HORIZONTAL',
};


export const PM_START_MESSAGE = 'Private conversation with';
export const BUSY_BROADCAST_THRESHOLD = 6;
