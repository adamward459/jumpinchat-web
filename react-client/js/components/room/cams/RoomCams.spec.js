import React from 'react';
import { shallow } from 'enzyme';
import { RoomCams } from './RoomCams.react';
import { layouts } from '../../../constants/RoomConstants';

describe('<RoomCams />', () => {
  let roomCams;
  beforeEach(() => {
    const props = {
      room: null,
      feeds: [],
      cams: [],
      canBroadcast: false,
      localStream: null,
      playYoutubeVideos: true,
      localAudioActive: false,
      showYoutubeVolume: false,
      showVolumeControl: false,
      currentlyPlaying: null,
      user: {
        operator_id: null,
        settings: {
          playYtVideos: true,
        },
      },
      layout: layouts.VERTICAL,
      globalVolume: 0,
    };

    roomCams = new RoomCams(props);
  });

  describe('getFeedsUpdated', () => {
    it('should return true if remote feed exists, new local feed added', () => {
      const previousFeeds = [{
        remoteFeed: {
          rfid: 'remote',
        },
      }];

      const newFeeds = [
        ...previousFeeds,
        {},
      ];

      expect(RoomCams.getFeedsUpdated(newFeeds, previousFeeds)).toEqual(true);
    });

    it('should return false if first feed added', () => {
      const previousFeeds = [];

      const newFeeds = [
        ...previousFeeds,
        {},
      ];

      expect(RoomCams.getFeedsUpdated(newFeeds, previousFeeds)).toEqual(false);
    });

    it('should return false if there is no change', () => {
      const previousFeeds = [{
        remoteFeed: {
          rfid: 'remote',
        },
      }];

      const newFeeds = [
        ...previousFeeds,
      ];

      expect(RoomCams.getFeedsUpdated(newFeeds, previousFeeds)).toEqual(false);
    });
  });

  describe('componentWillReceiveProps', () => {
    beforeEach(() => {
      roomCams._getCamDimensions = jest.fn();
    });

    it('should not get dimensions if there are no feeds in next props', () => {
      roomCams.componentWillReceiveProps({ ...roomCams.props, feeds: [] });
      expect(roomCams._getCamDimensions).not.toHaveBeenCalled();
    });

    it('should not get dimensions if current feeds === next feeds', () => {
      roomCams.props = {
        feeds: [
          {
            remoteFeed: { rfid: '123' },
          },
          {
            remoteFeed: { rfid: '321' },
          },
        ],
        globalVolume: 0,
      };
      roomCams.componentWillReceiveProps({
        feeds: [
          {
            remoteFeed: { rfid: '123' },
          },
          {
            remoteFeed: { rfid: '321' },
          },
        ],
      });
      expect(roomCams._getCamDimensions).not.toHaveBeenCalled();
    });

    it('should get cam dimensions if feeds have changed', () => {
      roomCams.props = {
        feeds: [
          {
            remoteFeed: { rfid: '123' },
          },
        ],
      };
      roomCams.componentWillReceiveProps({
        feeds: [
          {
            remoteFeed: { rfid: '123' },
          },
          {
            remoteFeed: { rfid: '321' },
          },
        ],
      });
      expect(roomCams._getCamDimensions).toHaveBeenCalled();
    });

    it('should get cam dimensions if currentlyPlaying changes', () => {
      roomCams.props = {
        feeds: [],
        currentlyPlaying: {
          mediaId: 'foo',
          startAt: '2019-01-29T22:46:26.356Z',
          endTime: '2019-01-29T22:46:26.356Z',
          duration: 123,
        },
      };
      roomCams.componentWillReceiveProps({ feeds: [], currentlyPlaying: null });
      expect(roomCams._getCamDimensions).toHaveBeenCalled();
    });
  });

  describe('getCamDimensions', () => {
    beforeEach(() => {
      roomCams.camwrapper = {
        offsetWidth: 10,
        offsetHeight: 10,
      };

      roomCams.camcontainer = {
        offsetWidth: 5,
        offsetHeight: 5,
      };

      roomCams.containerInternal = {
        offsetWidth: 15,
        offsetHeight: 15,
      };

      roomCams.pack = jest.fn();
      roomCams.setState = jest.fn();
    });

    it('should use current props if nextProps args have no feeds', () => {
      roomCams.props.feeds = [{}];
      roomCams._getCamDimensions();
      expect(roomCams.pack).toHaveBeenCalled();
    });

    it('should call pack with correct values', () => {
      roomCams.props.feeds = [{}, {}, {}];
      roomCams._getCamDimensions();
      expect(roomCams.pack).toHaveBeenCalledWith(3, 10, 15);
    });

    it('should set dimensions into state', () => {
      roomCams.setState = jest.fn();
      roomCams.props.feeds = [{}, {}, {}];
      roomCams.pack = jest.fn(() => ({
        width: 1,
        height: 2,
        x: 3,
        y: 4,
      }));
      roomCams._getCamDimensions();
      expect(roomCams.setState).toHaveBeenCalledWith({
        camDimensions: {
          width: 1,
          height: 2,
          x: 3,
          y: 4,
        },
        camWrapperHeight: 15,
        camContainerWidth: 10,
      });
    });

    it('should use cam wrapper dimensions if video is playing', () => {
      roomCams.setState = jest.fn();
      roomCams.props.feeds = [{}, {}, {}];
      roomCams.props.currentlyPlaying = {
        mediaId: 'foo',
        startAt: '2019-01-29T22:46:26.356Z',
        endTime: '2019-01-29T22:46:26.356Z',
        duration: 123,
      };
      roomCams.pack = jest.fn(() => ({
        width: 1,
        height: 2,
        x: 3,
        y: 4,
      }));
      roomCams._getCamDimensions();
      expect(roomCams.setState).toHaveBeenCalledWith({
        camDimensions: {
          width: 1,
          height: 2,
          x: 3,
          y: 4,
        },
        camWrapperHeight: 15,
        camContainerWidth: 10,
      });
    });
  });

  describe('render', () => {
    let props;
    beforeEach(() => {
      props = {
        localStream: {
          stream: {
            getAudioTracks: jest.fn(() => ([])),
          },
        },
        camBroadcast: true,
        feeds: [
          {
            optionsOpen: false,
            userId: 'foo',
          },
          {
            optionsOpen: false,
            userId: 'bar',
          },
          {
            optionsOpen: false,
            userId: 'baz',
          },
        ],
        cams: [
          { userId: 'foo', stream: {} },
          { userId: 'bar', stream: {} },
          { userId: 'baz', stream: {} },
        ],
        room: {
          name: 'room',
          users: [1, 2, 3, 4],
          settings: {
            modOnlyPlayMedia: false,
          },
          attrs: {
            owner: null,
          },
        },
        users: [],
        playYoutubeVideos: true,
        userCount: 4,
        localAudioActive: false,
        showYoutubeVolume: false,
        showVolumeControl: false,
        user: {
          operator_id: 'foo',
          hasChangedHandle: true,
          settings: {
            playYtVideos: true,
          },
        },
        currentlyPlaying: {
          mediaId: 'foo',
          duration: 123,
          startAt: 0,
          startTime: '2019-01-29T22:46:26.356Z',
          endTime: '2019-01-29T22:46:26.356Z',
        },
        broadcastRestricted: false,
        hasHandleChanged: false,
        layout: layouts.VERTICAL,
        modOnlyPlayMedia: false,
        chatColors: [],
        canPlayMedia: true,
        feedsHighDef: true,
        feedsMuted: false,
        camsDisabled: false,
        settingsOptionsOpen: false,
      };
    });

    describe('footer', () => {
      it('should not show cam footer in vertical layout', () => {
        const wrapper = shallow(<RoomCams {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
      });

      it('should show cam footer in horizontal layout', () => {
        props.layout = layouts.HORIZONTAL;
        const wrapper = shallow(<RoomCams {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
      });
    });

    describe('cams wrapper', () => {
      it('should set max width for more than 2 feeds', () => {
        const wrapper = shallow(<RoomCams {...props} />);

        wrapper.setState({
          camDimensions: { x: 1, y: 1 },
          camWrapperHeight: 75,
          camContainerWidth: 150,
        });

        expect(wrapper.find('.cams__Wrapper').props().style).toEqual({ maxWidth: '100px' });
      });

      it('should not have a style if less than 3 feeds', () => {
        const wrapper = shallow(<RoomCams {...props} />);
        expect(wrapper.find('.cams__Wrapper').props().style).toEqual(undefined);
      });

      it('should not have a max width if width < height', () => {
        const wrapper = shallow(<RoomCams {...props} />);

        wrapper.setState({
          camContainerHeight: 200,
          containerWidth: 75,
        });

        expect(wrapper.find('.cams__Wrapper').props().style).toEqual(undefined);
      });

      it('should render correct number of cam windows', () => {
        const wrapper = shallow(<RoomCams {...props} />);

        expect(wrapper.find('RoomCam').length).toEqual(3);
      });

      describe('YoutubeVideoContainer', () => {
        it('should render youtube container if currentlyPlaying has a value', () => {
          props = {
            ...props,
            currentlyPlaying: {
              duration: 123,
              mediaId: 'foo',
              startAt: 0,
              startTime: '2019-01-29T22:46:26.356Z',
              endTime: '2019-01-29T22:46:26.356Z',
            },
          };
          const wrapper = shallow(<RoomCams {...props} />);
          expect(wrapper.find('YoutubeVideoContainer').length).toEqual(1);
        });

        it('should not render youtube container if `playYoutubeVideos` is false', () => {
          props.user.settings.playYtVideos = false;
          const wrapper = shallow(<RoomCams {...props} />);
          expect(wrapper.find('YoutubeVideoContainer').length).toEqual(0);
        });

        it('should not render youtube container if currentlyPlaying has no value', () => {
          props = { ...props, currentlyPlaying: null };
          const wrapper = shallow(<RoomCams {...props} />);
          expect(wrapper.find('YoutubeVideoContainer').length).toEqual(0);
        });

        it('should send currentlyPlaying value to video container', () => {
          props = {
            ...props,
            currentlyPlaying: {
              duration: 123,
              mediaId: 'foo',
              startTime: '2019-01-29T22:46:26.356Z',
              startAt: 0,
              endTime: '2019-01-29T22:46:26.356Z',
            },
          };
          const wrapper = shallow(<RoomCams {...props} />);
          expect(wrapper.find('YoutubeVideoContainer').props().videoDetails.mediaId)
            .toEqual('foo');
        });
      });
    });
  });
});
