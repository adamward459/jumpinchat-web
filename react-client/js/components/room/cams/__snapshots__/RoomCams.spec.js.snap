// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomCams /> render footer should not show cam footer in vertical layout 1`] = `
<div
  className="cams"
>
  <RoomCamsHeader
    audioContext={null}
    audioPtt={true}
    broadcastRestricted={false}
    canBroadcast={false}
    feedCount={3}
    localAudioActive={false}
    localStream={
      Object {
        "stream": Object {
          "getAudioTracks": [MockFunction],
        },
      }
    }
    room={
      Object {
        "attrs": Object {
          "owner": null,
        },
        "name": "room",
        "settings": Object {
          "modOnlyPlayMedia": false,
        },
        "users": Array [
          1,
          2,
          3,
          4,
        ],
      }
    }
    userCount={4}
  />
  <div
    className="cams__ContainerInternal"
  >
    <YoutubeVideoContainer
      dimensions={
        Object {
          "height": 0,
          "width": 0,
        }
      }
      hasChangedHandle={true}
      onVideoEnd={[Function]}
      onVideoReady={[Function]}
      optionsOpen={false}
      showVolumeControl={false}
      userIsOperator={true}
      videoDetails={
        Object {
          "duration": 123,
          "endTime": "2019-01-29T22:46:26.356Z",
          "mediaId": "foo",
          "startAt": 0,
          "startTime": "2019-01-29T22:46:26.356Z",
        }
      }
      volume={1}
    />
    <div
      className="cams__Wrapper"
      id="cam-wrapper"
    >
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "foo",
          }
        }
        index={0}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "foo",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "bar",
          }
        }
        index={1}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "bar",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "baz",
          }
        }
        index={2}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "baz",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
    </div>
  </div>
</div>
`;

exports[`<RoomCams /> render footer should show cam footer in horizontal layout 1`] = `
<div
  className="cams"
>
  <RoomCamsHeader
    audioContext={null}
    audioPtt={true}
    broadcastRestricted={false}
    canBroadcast={false}
    feedCount={3}
    localAudioActive={false}
    localStream={
      Object {
        "stream": Object {
          "getAudioTracks": [MockFunction],
        },
      }
    }
    room={
      Object {
        "attrs": Object {
          "owner": null,
        },
        "name": "room",
        "settings": Object {
          "modOnlyPlayMedia": false,
        },
        "users": Array [
          1,
          2,
          3,
          4,
        ],
      }
    }
    userCount={4}
  />
  <div
    className="cams__ContainerInternal"
  >
    <YoutubeVideoContainer
      dimensions={
        Object {
          "height": 0,
          "width": 0,
        }
      }
      hasChangedHandle={true}
      onVideoEnd={[Function]}
      onVideoReady={[Function]}
      optionsOpen={false}
      showVolumeControl={false}
      userIsOperator={true}
      videoDetails={
        Object {
          "duration": 123,
          "endTime": "2019-01-29T22:46:26.356Z",
          "mediaId": "foo",
          "startAt": 0,
          "startTime": "2019-01-29T22:46:26.356Z",
        }
      }
      volume={1}
    />
    <div
      className="cams__Wrapper"
      id="cam-wrapper"
    >
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "foo",
          }
        }
        index={0}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "foo",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "bar",
          }
        }
        index={1}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "bar",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
      <RoomCam
        audioContext={null}
        dimensions={null}
        feed={
          Object {
            "optionsOpen": false,
            "userId": "baz",
          }
        }
        index={2}
        optionsOpen={false}
        streamData={
          Object {
            "stream": Object {},
            "userId": "baz",
          }
        }
        user={
          Object {
            "hasChangedHandle": true,
            "operator_id": "foo",
            "settings": Object {
              "playYtVideos": true,
            },
          }
        }
        users={Array []}
        videoEnabled={true}
      />
    </div>
  </div>
  <RoomCamsFooter
    camsDisabled={false}
    canPlayMedia={true}
    chatColors={Array []}
    feedsCount={3}
    feedsHighDef={true}
    feedsMuted={false}
    layout="HORIZONTAL"
    modOnlyPlayMedia={false}
    playYoutubeVideos={true}
    roomHasOwner={false}
    roomName="room"
    settingsOptionsOpen={false}
    user={
      Object {
        "hasChangedHandle": true,
        "operator_id": "foo",
        "settings": Object {
          "playYtVideos": true,
        },
      }
    }
  />
</div>
`;
